### PR TITLE
Replace checked-in ZIP with a dynamic dependency

### LINKED ISSUE #491: Replace checked-in ZIP with a dynamic dependency
Coming from https://github.com/opensearch-project/opensearch-build/issues/1750, replace checked-in ZIP with a dynamic dependency downloaded during bwc tests.

```
./sql/integ-test/src/test/resources/bwc/1.13.2.0/opendistro-sql-1.13.2.0.zip
```

### PR DESCRIPTION
**Pull request by penghuo**
### Description
Replace checked-in ZIP with a dynamic dependency
 
### Issues Resolved
#491 
 
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: integ-test/build.gradle [hunk 1]
```diff
@@ -188,6 +189,8 @@ task compileJdbc(type: Exec) {
 String bwcVersion = "1.13.2.0";
 String baseName = "sqlBwcCluster"
 String bwcFilePath = "src/test/resources/bwc/"
+String bwcOpenDistroPlugin = "opendistro-sql-1.13.2.0.zip"
+String bwcRemoteFile = 'https://d3g5vo6xdbdb9a.cloudfront.net/downloads/elasticsearch-plugins/opendistro-sql/opendistro-sql-1.13.2.0.zip'
```
Comment (jackiehanyang) [inline integ-test/build.gradle:193]: How do we maintain this zip file in cloudfront? Like if this package has merged a new bug fix commit, do we need to build the zip file locally then upload it to cloudfront? Then although it's not a checked-in dependency anymore, it's still not a dynamic dependency?

Comment (penghuo) [reply integ-test/build.gradle:193]: opendistro-sql-1.13.2.0.zip is the released latest opendistro SQL plugin. we didn't maintain opendistro-sql-1.13.2.0.zip

Comment (jackiehanyang) [reply integ-test/build.gradle:193]: if opendistro SQL plugin release a new version, do we need to replace this zip file on cloudfront with the one build with latest version?

**Review by joshuali925 — APPROVED** [review]
what about the opensearch-ml zip?

**Review by davidcui1225 — APPROVED** [review]

File: integ-test/build.gradle [hunk 2]
```diff
@@ -188,6 +189,8 @@ task compileJdbc(type: Exec) {
 String bwcVersion = "1.13.2.0";
 String baseName = "sqlBwcCluster"
 String bwcFilePath = "src/test/resources/bwc/"
+String bwcOpenDistroPlugin = "opendistro-sql-1.13.2.0.zip"
```
Comment (dblock) [inline integ-test/build.gradle:192]: Nit: I would use `bcwVersion` from above in the file name.

File: integ-test/build.gradle (hunk 1 above)
Comment (dblock) [reply integ-test/build.gradle:193]: What's the official URL for this .zip? I see in https://opendistro.github.io/for-elasticsearch-docs/docs/install/plugins/ that we reference the cloudfront URL. If that's how it should be, so be it, but I would use something like opendistrowhateverthatwas.org if that exists.

**Review by dblock — COMMENTED** [review]
LGTM

**Comment by penghuo** [comment]
@dblock I think plugin zip only available at cloudfront URL. I didn't aware opendistrowhateverthatwas.org link.

**Pull request merged**